### PR TITLE
Implement signature help overload UI and parameter description UI

### PR DESCRIPTION
**Pull request by tomoikey**
Release Notes:

- Improved ... #12909

Optionally, include screenshots / media showcasing your addition that can be included in the release notes.

### Or...

Release Notes:

- N/A

# Purpose
The SignatureHelp PR I previously submitted had some TODO left for overloaded functions. Since there is still time before the release, I have created the UI for overloaded functions. I used VSCode’s UI as a reference.

# Description
I implemented **VSCode-like** signature help UI.
I have attached the VSCode UI that I used as a reference below.

### Has a parameter description (VSCode)
image

### Has no parameter descrption (VSCode)
image

# Movie
https://github.com/user-attachments/assets/3701424c-b75f-4625-8626-e045d70a5adc

# ScreenShot
### Page 3 (active signature)
image

### Page 2
image  

### Page 1
image  

### Page 2 (active signature)
image  

### Page 1
image

### Page 2 (active signature)
image

### Page 1
image

## Example 4 (Non overloaded function)
image

### CODE REVIEW

**Review by SomeoneToIgnore — COMMENTED** [review]
I gave it a spin in a pair of real projects and here we go again.

Our favorite:
image
I propose to use a certain (e.g. Rust) real LSP response that produces this result, and write a test that ensures we do not show this pop-up ever again.

I find this layout odd: 
image
I would center the right part vertically, same as the left part, and made arrows ' hitbox (not the image itself) wider, so I could click them simper without aiming much.

Here's another example:

image

this one is especially interesting code-wise, as reproduces another bug: 3/5 of the results are the same `selectors: K` string so seems that the results are not deduplicated?

Consider what other editors do:

VSCode
image

Intellij
image

Fleet
image

I sort of like VSCode's arrow shapes because they are wider, but we can just increase hitboxes and make it as useful.
Not sure about both parts' contents alignment though — in Zed that would look odd, I think?

I'm not 100% certain that I would like `fn foo` and the return type in this pop-up, but the idea looks interesting.

The large pop-up with all results combined and the scroll looks quite odd, so I'd avoid that design.

During testing, I've found a flickering issue that already exists in the codebase:
https://github.com/user-attachments/assets/b047f047-06ca-4956-afc3-6734037f7888
it looks like we do not clear the old state after the selections change for a frame or two?
Clearing it always would also cause flickering when clicking within the same bracket pair though.

File: crates/project/src/lsp_command/signature_help.rs [hunk 1]
```diff
@@ -15,98 +16,109 @@ pub const SIGNATURE_HELP_HIGHLIGHT_CURRENT: MarkdownHighlight =
...
+                        lsp::ParameterLabel::LabelOffsets(offset) => signature_information
+                            .label
+                            .chars()
+                            .skip(offset[0] as usize)
+                            .take((offset[1] - offset[0]) as usize)
+                            .collect::<String>(),
+                    };
+                    let label_length = label.len();
+
+                    let documentation =
```
Comment (SomeoneToIgnore) [inline crates/project/src/lsp_command/signature_help.rs:67]: This is wrong to consider any text as a markdown, as the latter implies specific char sequences to control its formatting, e.g. `*text*` rendered as plaintext vs *text* rendered as markdown.
So it's not correct to always do 
https://github.com/zed-industries/zed/blob/3666ba6a6ed9683c0907dcef73acc1e57206a042/crates/editor/src/signature_help.rs#L204-L209
as this PR does now.

We can use https://github.com/zed-industries/zed/blob/3666ba6a6ed9683c0907dcef73acc1e57206a042/crates/language/src/buffer.rs#L244 and render the text as a plain label in non-markdown cases.

Note that text could come from two places of the LSP type: 
* `lsp::Documentation::String`
* `lsp::Documentation::MarkupContent` for `MarkupKind::PlainText`

**Review by SomeoneToIgnore — COMMENTED** [review]
Got a pass over the code too.

**Comment by tomoikey** [comment]
Sorry for the delayed response.
I’m currently undergoing treatment for an illness, so my replies may be slow.

**Comment by SomeoneToIgnore** [comment]
No worries at all, I am not the fastest to reply too.
Please, take care and get well, all this can wait — we might close the PR if it gets over a month old to keep the queue cleaner, but please reopen that later then.

**Pull request closed without merging**